api/src/lib/form-handler.js

```diff
@@ -35,7 +35,7 @@ module.exports = (
       const render = async e => {
         const viewContext = getView(user, client, request, e);
         const template = await themeService.renderThemedTemplate(templateName, viewContext, request.query.client_id);
-        return reply(template);
+        return reply(template).code(e && e.output && e.output.statusCode || 200);
```

Comment:
I couldn't see a good way to surface validation errors to the user. Did this so at least we can test and log it as something other than a success.

Reply:
One thing I'm interested in play with is the use of dom-testing-library (the parent of react-testing-library) to test rendered html templates for stuff. Wanna experiment with that?